install_config/router/customized_haproxy_router.adoc

```diff
@@ -13,48 +13,673 @@ toc::[]
+The template file is a
+link:http://golang.org/pkg/text/template/[*golang* template].
+When processing the template, the router has access to {product-title}
+information including the router's deployment configuration,
+the set of admitted routes and some helper functions.
+
+When the router pod starts and every time it reloads, it creates a HAProxy
+configuration file and then it starts HAProxy.  The
+link:https://cbonte.github.io/haproxy-dconv/configuration-1.5.html[Haproxy configuration manual]
+describes all of the features of HAProxy and how to construct a valid
+configuration file.
+
+A *ConfigMap* can be used to add the new template to the router pod. With
+this approach, the router deployment configuration is modified to mount the
+xref:using-configmap-replace-template[*configMap*] as a volume in the router
+pod.  The `TEMPLATE_FILE` environment variable is set to the full path name
+of the template file in the router pod.
+
+An alternative approach is to modify the *haproxy-template.config* file
+and xref:rebuilding-your-router[rebuild] the router image. The new image
+is pushed to the the cluster repository and the routers deployment
+configuration *image:* field is updated with the new name. When the cluster is
+updated the image needs to be rebuilt and pushed.
+
+In either case the router pod starts with the template file.
 
 [[obtaining-router-configuration-template]]
 == Obtaining the Router Configuration Template
 
-You can obtain a new *_haproxy-config.template_* file from the latest router
-image by running:
+The HAProxy template file is fairly large and complex. For some changes, it may
+be easier to modify the existing template rather than writing a complete
+replacement.  You can obtain a *_haproxy-config.template_* file from a running
+router by running this on master, referencing the router pod:
+
+----
+# oc get po
+NAME                       READY     STATUS    RESTARTS   AGE
+router-2-40fc3             1/1       Running   0          11d
+# oc rsh router-2-40fc3 cat haproxy-config.template > haproxy-config.template
+# oc rsh router-2-40fc3 cat haproxy.config > haproxy.config
+----
+
+or by logging onto the node that is running the router:
 
 ----
 ifdef::openshift-enterprise[]
 # docker run --rm --interactive=true --tty --entrypoint=cat \
-    registry.access.redhat.com/openshift3/ose-haproxy-router:v3.0.2.0 haproxy-config.template
+    registry.access.redhat.com/openshift3/ose-haproxy-router:$version haproxy-config.template
 endif::[]
 ifdef::openshift-origin[]
 # docker run --rm --interactive=true --tty --entrypoint=cat \
     openshift/origin-haproxy-router haproxy-config.template
 endif::[]
 ----
 
+Where the image name is from "docker images"
+
 Save this content to a file for use as the basis of your customized template.
+The saved haproxy.config shows what is actually running.
+
+
+[[router-configuration-template]]
+== Modifying the Router Configuration Template
+
+[[router-template-background]]
+
+=== Background
```

Comment:
I am not sure we should be restating, the golang documentation here. Why is the link to to golang documentation and writing up our additions not sufficient?